### PR TITLE
Node Explorer: Include full DNSName for external nodes

### PR DESCRIPTION
**Pull request by tylersmalley**
Nodes that are external to a user's Tailnet, or "Shared In" should be identified by having the rest of the DNS name set as a description.

Additionally, we are displaying the name according to the DNS entry, not the OS's hostname.

Closes ENG-1919
Stacked on https://github.com/tailscale-dev/vscode-tailscale/pull/147

image

### CODE REVIEW

File: tsrelay/handler/get_serve.go [hunk 1]
```diff
@@ -40,16 +40,22 @@ type currentTailnet struct {
 	MagicDNSEnabled bool
 }
 
+type selfStatus struct {
+	peerStatus
+	CurrentTailnet currentTailnet
```
Comment (marwan-at-work) [inline tsrelay/handler/get_serve.go:45]: This should actually be analogous to the go type, and not be defined on `self` but on the top level status, because it has nothing to do with your peers or yourself, both of you are under the same tailnet.

File: tsrelay/handler/get_serve.go [hunk 2]
```diff
@@ -40,16 +40,22 @@ type currentTailnet struct {
...
+}
```
[the remark this answers is not in the document]
Comment (tylersmalley) [reply tsrelay/handler/get_serve.go:46]: `tailscale status --json` does not include a `CurrentTailnet` for peers. We could make it optional, but it shouldn't be optional inside Self. Do the Go client types suggest otherwise?

File: src/types.ts [hunk 3]
```diff
@@ -21,18 +21,21 @@ export interface Handlers {
...
   ID: string;
+  ServerName: string;
   HostName: string;
   Active?: boolean;
+  IsExternal: boolean;
   Online?: boolean;
   TailscaleIPs: string[];
   sshHostKeys?: string[];
   ShareeNode?: boolean;
-  TailnetName: string;
```
Comment (marwan-at-work) [inline src/types.ts:30]: Out of curiosity, why was this removed? it breaks `createTsUri` above

File: src/node-explorer-provider.ts [hunk 4]
```diff
@@ -310,18 +327,21 @@ export class PeerTree extends PeerBaseTreeItem {
   public ID: string;
   public HostName: string;
   public TailscaleIPs: string[];
-  public TailnetName: string;
```
Comment (marwan-at-work) [inline src/node-explorer-provider.ts:313]: out of curiosity why was this removed? It breaks `createTsUri` above

**Review by marwan-at-work — APPROVED** [review]
🎉
